### PR TITLE
Category related code

### PR DESCRIPTION
**Pull request by MilianCode**

### CODE REVIEW

File: src/main/java/com/miliancode/controller/BookController.java [hunk 1]
```diff
@@ -35,7 +36,9 @@ public class BookController {
     @Operation(summary = "Get all books", description = "Get a list of all "
             + "avalaible books")
     @PreAuthorize("hasAnyRole('USER', 'ADMIN')")
-    public List<BookDto> getAll(@ParameterObject @PageableDefault Pageable pageable) {
+    public List<BookDtoWithoutCategotyIds> getAll(
```
Comment (Elena-Bruyako) [inline src/main/java/com/miliancode/controller/BookController.java]: revert to BookDto

File: src/main/java/com/miliancode/model/Book.java [hunk 2]
```diff
@@ -34,4 +42,11 @@ public class Book {
     private String coverImage;
     @Column(name = "is_deleted", nullable = false)
     private boolean isDeleted = false;
+    @ManyToMany(fetch = FetchType.LAZY)
```
Comment (Elena-Bruyako) [inline src/main/java/com/miliancode/model/Book.java]:
```suggestion
    @ManyToMany
```

File: src/main/java/com/miliancode/service/impl/CategoryServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+@RequiredArgsConstructor
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+    private final CategoryMapper mapper;
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto requestDto) {
+        Category category = mapper.toModel(requestDto);
+        Category savedCategory = categoryRepository.save(category);
+        return mapper.toDto(savedCategory);
```
Comment (Elena-Bruyako) [inline src/main/java/com/miliancode/service/impl/CategoryServiceImpl.java]:
```suggestion
        return mapper.toDto(categoryRepository.save(category));
```

File: src/main/java/com/miliancode/service/impl/CategoryServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Override
+    public CategoryDto update(Long id, CreateCategoryRequestDto requestDto) {
+        Category category = mapper.toModel(requestDto);
+        category.setId(id);
```
Comment (Elena-Bruyako) [inline src/main/java/com/miliancode/service/impl/CategoryServiceImpl.java]:
```suggestion
        Category category = repository.findById(id)
                .orElseThrow(()....
        mapper.updateCategoryFromDto(.....
        return mapper.toDto(categoryRepository.save(category));
```

File: src/main/java/com/miliancode/service/impl/CategoryServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    private final CategoryMapper mapper;
+
+    @Override
+    public CategoryDto save(CreateCategoryRequestDto requestDto) {
+        Category category = mapper.toModel(requestDto);
+        Category savedCategory = categoryRepository.save(category);
+        return mapper.toDto(savedCategory);
+    }
+
+    @Override
```
Comment (Elena-Bruyako) [inline src/main/java/com/miliancode/service/impl/CategoryServiceImpl.java:29]: Check where we need Transactional

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by sarakhmen — APPROVED** [review]

**Pull request merged**